common/check_transfer_queues_status

```diff
@@ -1,85 +1,65 @@
- WHEN state = 'N' THEN 'queues.requests.nosources.' || replace(activity, ' ', '_') || '.' || nvl(replace(substr(external_host, 9, length(external_host)-13), '.', '_'), 'no_fts_host')
- WHEN state = 'O' THEN 'queues.requests.onlytapesources.' || replace(activity, ' ', '_') || '.' || nvl(replace(substr(external_host, 9, length(external_host)-13), '.', '_'), 'no_fts_host')
- WHEN state = 'A' THEN 'queues.requests.submissionfailed.' || replace(activity, ' ', '_') || '.' || nvl(replace(substr(external_host, 9, length(external_host)-13), '.', '_'), 'no_fts_host')
- WHEN state = 'U' THEN 'queues.requests.suspend.' || replace(activity, ' ', '_') || '.' || nvl(replace(substr(external_host, 9, length(external_host)-13), '.', '_'), 'no_fts_host')
- WHEN state = 'P' THEN 'queues.requests.preparing.' || replace(activity, ' ', '_') || '.' || nvl(replace(substr(external_host, 9, length(external_host)-13), '.', '_'), 'no_fts_host')
- ELSE state
-END state_desc,
-num_rows
-FROM
-(
-select state, count(*) num_rows, activity, external_host
-FROM {schema}requests
-GROUP BY state, activity, external_host
-)""".format(schema=schema)
-
-PROM_SERVERS = config_get('monitor', 'prometheus_servers', raise_exception=False, default='')
-if PROM_SERVERS != '':
-    PROM_SERVERS = PROM_SERVERS.split(',')
-
 if __name__ == "__main__":
     try:
-        registry = CollectorRegistry()
-        g = Gauge('conveyor_queues_requests', '', labelnames=('state', 'activity', 'external_host'), registry=registry)
         session = get_session()
-        for k in session.execute(active_queue).fetchall():
-            print(k[0], k[1], end=" ")
-            probe_metrics.gauge(name=k[0].replace('-', '_')).set(k[1])
-            items = k[0].split('.')
-            state = items[2]
-            activity = items[3]
-            external_host = items[4].replace('-', '_')
-            g.labels(**{'activity': activity, 'state': state, 'external_host': external_host}).set(k[1])
-        if len(PROM_SERVERS):
-            for server in PROM_SERVERS:
-                try:
-                    push_to_gateway(server.strip(), job='check_transfer_queues_status', registry=registry)
-                except:
-                    continue
-    except:
+        statement = select(
+            models.Request.state,
+            models.Request.activity,
+            models.Request.external_host,
+            func.count()
+        ).group_by(
+            models.Request.state,
+            models.Request.activity,
+            models.Request.external_host
+        )
+
+        with PrometheusPusher() as manager:
+            for row in session.execute(statement):
+                state = f"queues.requests.{row.state.name.lower()}"
+                activity = row.activity.replace(" ", "_")
+                if row.external_host is not None:
+                    external_host = urlparse(row.external_host).netloc
+                    external_host = external_host.split(':')[0].replace(".", "_")
```

Comment:
Why not `external_host = urlparse(row.external_host).hostname.replace(".", "_")`?

Reply:
Because I misread the docs 🤦